### PR TITLE
Fix: Prevent automatic parsing of string data types to numbers

### LINKED ISSUE #36149: Reduce transform causes loss of precision in 64 bit series names
**What happened**: When using a reduce transform on a time series with series names made of decimal representation of unsigned 64bit numbers (e.g. the string '2882377905688543293'), the resulting field column will truncate the precision of the represented number(2882377905688543000). I am assuming something is needlessly parsing to a numeric representation that is insufficient for the uint64.

**What you expected to happen**: The series name string should be duplicated verbatim in the reduced output.

**How to reproduce it (as minimally and precisely as possible)**:
 - Create an influx query on a measurement with one of the tags holding 64bit unsigned integers. Group by time and tag.
 - Alias the series name by tag name "$tag_My64BitTag"
 - Add a reduce transformation that calculates stats on each series.
 - The series name will be transformed in the reduced output. e.g. '2882377905688543293' becomes 2882377905688543000
 - Adding some non numeric characters to the series alias, "$tag_My64BitTag foo" cures the loss of precision, but isn't acceptable for my use case. Presumably that is causing the attempt to parse to fail.

**Environment**:
- Grafana version: 7.4.3

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

This changes the displayProcessor to prevent parsing `FieldType.string` fields as numbers, to fix a bunch of cases where that's not desired (leading zeroes being dropped, [very big numbers being truncated](https://github.com/grafana/grafana/issues/36149), etc) by setting the `unit` to `string` if a `FieldType.string` field doesn't have an explicitly set `unit`.

You can opt into the previous automatic parsing behaviour by setting the unit to `none` (which I have renamed to "Automatic")

**Which issue(s) this PR fixes**:

Fixes #36149

**Special notes for your reviewer**:

 - Should we migrating existing panels with no selected unit to the Automatic/none unit?
 - Do we want to change what the Automatic/none unit does?
 - Is this actually a breaking change? Would migrating existing panels change whether its breaking or not?
 
# Release notice breaking change

Panels will no longer automatically try to parse fields with data type `string` to numbers to prevent issues where values that _looked_ like numbers could be changed inadvertently (for example, leading zeroes getting dropped). Any panels that relied on this old behaviour should update to explicitly add an appropriate unit, or use the "Misc / none" unit.

### CODE REVIEW

**Comment by torkelo** [comment]
While I do want this really badly, defaulting to parsing strings to numbers is causing lots issues. This is tricky to change as it has the potential to break many thousands of dashboards that rely on this behavior.

File: packages/grafana-data/src/field/displayProcessor.ts [hunk 1]
```diff
@@ -67,6 +67,8 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
     if (!isBooleanUnit(unit)) {
       unit = 'bool';
     }
+  } else if (!unit && field.type === FieldType.string) {
+    unit = 'string';
```
Comment (torkelo) [inline packages/grafana-data/src/field/displayProcessor.ts:71]: so to get the default behavior you have to set a unit, but we do not have a "no unit unit", so you would be forced to set it to short or something

Comment (joshhunt) [reply packages/grafana-data/src/field/displayProcessor.ts:71]: we do appear to have a "none" unit?

1018_2022-03-01-16-21_chrome

Comment (torkelo) [reply packages/grafana-data/src/field/displayProcessor.ts:71]: ah, that's true! forgot about that one!

**Comment by joshhunt** [comment]
@grafanabot nooo!!

File: packages/grafana-data/src/valueFormats/categories.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const getCategories = (): ValueFormatCategory[] => [
   {
     name: 'Misc',
     formats: [
-      { name: 'none', id: 'none', fn: toFixedUnit('') },
+      { name: 'Automatic', id: 'none', fn: toFixedUnit('') },
```
Comment (torkelo) [inline packages/grafana-data/src/valueFormats/categories.ts:41]: Not sure about this, it's not really automatic.

Comment (torkelo) [inline packages/grafana-data/src/valueFormats/categories.ts:41]:
```suggestion
      { name: 'None', id: 'none', fn: toFixedUnit('') },
```

**Review by torkelo — APPROVED** [review]
Not sure about the naming of "Automatic" as there is no automatic unit, only automatic parsing of strings. 

Other options
* None
* Plain
* Number

**Review by torkelo — COMMENTED** [review]
> Should we migrating existing panels with no selected unit to the Automatic/none unit?

Tricky, not sure we need to but we might have to. So it's more a question if we should wait for feedback/escalations to know for sure. 

> Do we want to change what the Automatic/none unit does?

We could, but then we would need to introduce a new one named "number" maybe?

File: packages/grafana-data/src/valueFormats/categories.ts (hunk 2 above)
Comment (joshhunt) [reply packages/grafana-data/src/valueFormats/categories.ts:41]: Fair!

**Comment by joshhunt** [comment]
Thanks @torkelo. Going to merge this in, renaming the `none` unit to "Number". 

I'm not sure _what_ changes we would make to this none/number unit display processor, but i was just floating it as a possibility seeing as we're making this change.

Will await feedback to consider whether we need a migration or not.

**Pull request merged**